### PR TITLE
Add an example file and test for using mock-inspect with nodejs

### PR DESCRIPTION
**Pull request by aychtang**

### CODE REVIEW

**Comment by aychtang** [comment]
Hey guys, was looking at potentially adding some examples like I mentioned in #17.

There is an error with the example test I added:
```
 FAIL  examples/nodejs/nodeJsExample.spec.ts
  ● should send a request to https://example.com

    Unhandled error. (Error: [MSW] Error: captured a GET https://example.com/ request without a corresponding request handler.

      If you wish to intercept this request, consider creating a request handler for it:

      rest.get('https://example.com/', (req, res, ctx) => {
        return res(ctx.text('body'))
      })

      at onUnhandledRequest (node_modules/msw/node/index.js:2081:19)
      at Object.<anonymous> (node_modules/msw/node/index.js:2155:25)
```

**Comment by aychtang** [comment]
It doesn't seem to correctly mock the call in `examples/nodejs/index.ts`

**Comment by aychtang** [comment]
Ok fixed it in d79931d101ebb02542a70cb39c8d73e7634be439, it's just that in the mock request I was including `www.` in the `requestPattern`.

The github action is failing as the new test is not under rootDir, where do we want to put this stuff? It shouldn't be in `src` as it's not source code for the app I think.

File: examples/nodejs/nodejs.spec.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        // You can assert that the request has not been made in this test:
+        let expectedErrorWhenRequestNotMade
+        try {
+            mockedRequest.expectRequestToHaveBeenMade()
+        } catch (error) {
+            expectedErrorWhenRequestNotMade = error
+        }
+        expect(expectedErrorWhenRequestNotMade.message).toMatch(
+            "You expected that the request has been made, but it was not."
+        )
```
Comment (aychtang) [inline examples/nodejs/nodejs.spec.ts:23]: Just an observation that this part of the API is a bit clunky, think we should be able to do this better with a `.toThrow(expectedError)` if we set it up better

Comment (aychtang) [reply examples/nodejs/nodejs.spec.ts]: Ignore me, took this example from the lower level tests

**Comment by thomaschaplin** [comment]
@aychtang what's the status of this PR? Looks like the unit test pipeline is failing

**Comment by aychtang** [comment]
I will add some kind of example as a separate repo which we can then link to from the README of the main project.

**Pull request closed without merging**